File: platformer/src/main/java/com/mtgames/platformer/entities/AutoScroll.java
package com.mtgames.platformer.entities;

import com.mtgames.platformer.gfx.Screen;
import com.mtgames.platformer.scripting.JythonFactory;
import com.mtgames.platformer.scripting.interfaces.EntityInterface;
import com.mtgames.platformer.settings.Properties;
import com.mtgames.utils.Debug;

public class AutoScroll extends AdvancedEntity {

	private final EntityInterface entityInterface;

	public AutoScroll(int x, int y, Properties properties) {
		super(properties, x, y);

		entityInterface = (EntityInterface) JythonFactory.getJythonObject("com.mtgames.platformer.scripting.interfaces.EntityInterface", "entities/AutoScroll.py");
		entityInterface.init(this);
	}

	public void tick() {
		entityInterface.tick(this);
	}

	public void render(Screen screen) {
		entityInterface.render(this, screen);
	}
}

File: platformer/src/main/resources/python/entities/autoScroll.py
from com.mtgames.platformer.scripting.interfaces import EntityInterface

class AutoScroll(EntityInterface):
    def __init__(self):
       self.speed = 0

    def init(self, entity):
        self.speed = entity.getProperties().getSpeed()

    def tick(self, entity):
        entity.x += self.speed

        entity.y = entity.getProperties().getLevel().entities.get(1).y

    def render(self, entity):
        pass
